src/lib/utils.js

```diff
@@ -125,7 +125,7 @@ export function escapeRegExp(s) {
 }
 
 export function defaultProgress(status) {
-  const seconds = Math.trunc(Date.now() / 1000);
+  const seconds = Math.trunc(Date.now());
```

Comment:
It was better to compare between parallelized and sequential metadata decryption.